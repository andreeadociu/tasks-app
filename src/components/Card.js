import React from "react";
import CardHeader from "./CardHeader";

const Card = (props) => {
  return (
    <div data-testid="card">
      <div className="col s12 m6 l3">
        <div className="card blue-grey darken-1">
          <div className="card-content white-text">
            <CardHeader
              title={props.title}
              id={props.id}
              onCompletePress={props.onComplete}
              status={props.status}
            />
            <p>{props.content}</p>
            <p className="status">Status: {props.status}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Card;
