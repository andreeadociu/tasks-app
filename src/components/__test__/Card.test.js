import React from "react";
import ReactDOM from "react-dom";
import Card from "../Card";
import { BrowserRouter as Router } from "react-router-dom";

import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Router>
      <Card />
    </Router>,
    div
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Router>
        <Card
          title="Test Task"
          content="this is a test task"
          status="completed"
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
