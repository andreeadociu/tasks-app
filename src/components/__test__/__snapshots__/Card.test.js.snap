// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="card"
>
  <div
    className="col s12 m6 l3"
  >
    <div
      className="card blue-grey darken-1"
    >
      <div
        className="card-content white-text"
      >
        <div
          className="card-action"
        >
          <span
            className="card-title"
          >
            Test Task
          </span>
          <a
            href="/edit/undefined"
            onClick={[Function]}
          >
            Edit
          </a>
          <a
            className="grey-text"
            status="completed"
          >
            Complete
          </a>
        </div>
        <p>
          this is a test task
        </p>
        <p
          className="status"
        >
          Status: 
          completed
        </p>
      </div>
    </div>
  </div>
</div>
`;
